extensions/amp-date-picker/0.1/amp-date-picker.js

```diff
@@ -98,47 +95,90 @@ const attributesToForward = [
   'number-of-months',
   'orientation',
   'reopen-picker-on-clear-date',
-  'required',
-  'screen-reader-input-message',
-  'show-clear-date',
-  'show-clear-dates',
-  'show-default-input-icon',
   'with-full-screen-portal',
   'with-portal',
   'day-size',
-  'week-day-format',
   'month-format',
 ];
 
-const DEFAULT_DATE_SIZE = 45; // px
+/** @enum {string} */
+const DatePickerMode = {
+  STATIC: 'static',
+  OVERLAY: 'overlay',
+};
+
+/** @enum {string} */
+const DatePickerState = {
+  OVERLAY_CLOSED: 'overlay-closed',
+  OVERLAY_OPEN_INPUT: 'overlay-open-input',
+  OVERLAY_OPEN_PICKER: 'overlay-open-picker',
+  STATIC: 'static',
+};
+
+/** @enum {string} */
+const DatePickerType = {
+  SINGLE: 'single',
+  RANGE: 'range',
+};
+
+/** @enum {string} */
+const DateFieldType = {
+  DATE: 'date',
+  START_DATE: 'start-date',
+  END_DATE: 'end-date',
+};
+
+/**
+ * The size in PX of each calendar day. This value allows the date picker to
+ * fit within a 320px wide viewport when fully rendered.
+ */
+const DEFAULT_DATE_SIZE = 39;
 
 const DEFAULT_FIRST_DAY_OF_WEEK = 0; // Sunday
 
 const DEFAULT_WEEK_DAY_FORMAT_CSS = 'i-amphtml-default-week-day-format';
 
+const DEFAULT_WEEK_DAY_FORMAT = 'dd';
+
+const INPUT_FOCUS_CSS = 'amp-date-picker-selecting';
+
 class AmpDatePicker extends AMP.BaseElement {
   /** @param {!AmpElement} element */
   constructor(element) {
     super(element);
 
+    this.timer_ = Services.timerFor(this.win);
+
     /** @private @const */
     this.ampdoc_ = Services.ampdoc(this.element);
 
     /** @private @const */
     this.win_ = this.ampdoc_.win;
 
+    /** @private @const */
+    this.document_ = this.element.ownerDocument;
```

Comment:
this.ampdoc_.getRootNode();

Reply:
Unfortunately you can't call `createElement` on `ShadowRoot`